src/index.js

```diff
@@ -1,59 +1,129 @@
 // import _ from 'lodash';
 
+import form, { todoInput, doList } from './modules/do.js';
+
 // eslint-disable-line
 
 import './style.css';
 
-const dolist = [
-  {
-    index: 0,
-    description: 'Doing some exercise',
-    completed: true,
-  },
-  {
-    index: 1,
-    description: 'Meeting with family',
-    completed: true,
-  },
-  {
-    index: 2,
-    description: 'Watching football games',
-    completed: true,
-  },
-  {
-    index: 3,
-    description: 'Study',
-    completed: true,
-  },
-];
-
-const taskInput = document.querySelector('.do-list');
-
-const lists = () => {
-  const list = dolist.map((items) => `<div class="item">
-        <div class="btn-check">
-        <i class="fa-regular fa-square"></i>
-        </div>
-        <p>${items.description}</p>
-        <div class="task-dots">
-        <i class="fa-solid fa-ellipsis-vertical"></i>
-        <i class="fa-solid fa-trash"></i>
-        </div>
-    </div>`).join('');
-
-  taskInput.insertAdjacentHTML('afterend', list);
+// Vars
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let editTodoId = -1;
+
+// render todos
+
+const renderTodos = () => {
+  if (todos.length === 0) {
+    doList.innerHTML = '<center>Nothing To Do!</center>';
+    return;
+  }
+
+  // clear element before re render
+  doList.innerHTML = '';
+
+  todos.forEach((todo, index) => {
+    doList.innerHTML += `
+          <div class="item" id=${index}>
+                  <i class="fa-regular ${todo.completed ? 'fa-square-check' : 'fa-square'}"
+                  data-action='check'
+                  ></i>
+                  <p data-action='edit'>${todo.description}</p>
+                  <i class="fa-solid fa-trash" data-action='delete'>                <i class="fa-solid fa-ellipsis-vertical" data-action='move'></i>
+                  </i>
+          </div>`;
+  });
+};
+
+// Save todo
+
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+
+  // check if the todolist is empty
+  const isEmpty = todoValue === '';
+
+  if (isEmpty) {
+    alert('To do list is empty');
+  } else if (todos.some((todo) => todo.description.toUpperCase() === todoValue.toUpperCase())) {
+    alert('Todo already exists!');
+  } else {
+    if (editTodoId >= 0) {
+      // update the edit todo
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        description: index === editTodoId ? todoValue : todo.description,
+      }));
+
+      editTodoId = -1;
+    } else {
+      todos.push({
+        indexNum: `@${Math.floor(Math.random() * todos.length)}`,
+        description: todoValue,
+        completed: false,
+      });
+    }
+    todoInput.value = '';
+  }
 };
```

Comment:
- [ ] The project requirement demanded that the new functionality should be implemented in a new js file(module). Kindly move these functions to modules and import them where needed.